### PR TITLE
[JENKINS-71038] CSP compatibility: PluginManager/installed.jelly

### PR DESCRIPTION
**Pull request by yaroslavafenkin**
See [JENKINS-71038](https://issues.jenkins.io/browse/JENKINS-71038).

### Testing done
Tested manually whether the behaviour is the same before and after the fix.

### Proposed changelog entries
- Improve CSP compatibility by un-inlining javascript code

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [x] ~New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.~
- [x] ~New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.~
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [x] ~For dependency updates, there are links to external changelogs and, if possible, full differentials.~
- [x] ~For new APIs and extension points, there is a link to at least one consumer.~

### Desired reviewers

@jenkinsci/core-security-review 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7912"></a>

### CODE REVIEW

File: core/src/main/resources/hudson/PluginManager/_installed.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      btn.getAttribute("url") +
+        "/make" +
+        (btn.checked ? "Enabled" : "Disabled"),
+      {
+        method: "post",
+        headers: crumb.wrap({}),
+      }
+    ).then((rsp) => {
+      if (!rsp.ok) {
+        $("needRestart").innerHTML = rsp.responseText;
```
Comment (yaroslavafenkin) [inline core/src/main/resources/hudson/PluginManager/_installed.js:28]: `responseText` does not exist on Response object as per https://developer.mozilla.org/en-US/docs/Web/API/Response#instance_properties. I'm unsure what was the intent here.

Comment (timja) [reply core/src/main/resources/hudson/PluginManager/_installed.js]: my mistake, I pushed to your PR in https://github.com/jenkinsci/jenkins/pull/7912/commits/9aa54268f32dd45e0363655bef5a71bdeeb9bad6

**Review by timja — APPROVED** [review]

**Comment by yaroslavafenkin** [comment]
> The last merge with the master branch seems incorrect

Yeah looks wrong, I'll check it later today.

**Review by NotMyFault — APPROVED** [review]

**Comment by basil** [comment]
Merge still looks wrong, as it contains `btn = Event.element(o)` rather than `btn = o.target`.

**Comment by yaroslavafenkin** [comment]
> Merge still looks wrong, as it contains `btn = Event.element(o)` rather than `btn = o.target`.

Not a matter of merge, it's that my change is from ancient times when prototype was still there. I've removed the usages in https://github.com/jenkinsci/jenkins/pull/7912/commits/b83d0e9d2e037e79f712a156ef29b5335bdff807 and retested manually whether everything is ok.

**Comment by basil** [comment]
> Not a matter of merge

Was the bug not introduced in the **merge** commit 86a046e2ef, which pulled in commit 82afbc4768 (where `btn = Event.element(o)` was changed to `btn = o.target`) for the first time in this branch without updating the `.js` file?

**Comment by yaroslavafenkin** [comment]
> Was the bug not introduced in the **merge** commit [86a046e](https://github.com/jenkinsci/jenkins/commit/86a046e2ef89abc0b915f5270e2d51f8bdaa4872), which pulled in commit [82afbc4](https://github.com/jenkinsci/jenkins/commit/82afbc4768527ca7dbdca78df0dbd6b451766a4f) (where `btn = Event.element(o)` was changed to `btn = o.target`) for the first time in this branch without updating the `.js` file?

Yeah, it was. I wasn't fully aware of what I'm pulling in and missed updating the `.js` file. 
I took 'merge looks wrong' as incorrectly resolved merge conflict at first, but I get your point now.

File: core/src/main/resources/hudson/PluginManager/_installed.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+(function () {
+  const restartRequired = document
+    .querySelector("#is-restart-required-for-completion")
```
Comment (basil) [inline core/src/main/resources/hudson/PluginManager/_installed.js:3]: [nit] It does not matter one way or another, but `getElementById` is more common in this codebase to select an element by its ID (110 usages as opposed to 29 usages of `querySelector("#`). It also performs a little better in theory, but not enough to matter in practice. No strong preference one way or another, so feel free to ignore this comment if you want.

```suggestion
    .getElementById("is-restart-required-for-completion")
```

File: core/src/main/resources/hudson/PluginManager/_installed.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        method: "post",
+        headers: crumb.wrap({}),
+      }
+    ).then((rsp) => {
+      if (!rsp.ok) {
+        rsp.text().then((responseText) => {
+          document.getElementById("needRestart").innerHTML = responseText;
+        });
+      }
+      updateMsg(btn);
```
Comment (basil) [inline core/src/main/resources/hudson/PluginManager/_installed.js:32]: I can't see any particular reason to pass `btn` to `updateMsg`. `updateMsg` is not declared to accept an argument, nor does it use any arguments.

```suggestion
      updateMsg();
```

**Review by basil — APPROVED** [review]
Thanks for the PR!

File: core/src/main/resources/hudson/PluginManager/_installed.js (hunk 3 above)
Comment (yaroslavafenkin) [reply core/src/main/resources/hudson/PluginManager/_installed.js:32]: Whoops, this is actually a development leftover. Will remove, thanks for pointing this out.

**Review by basil — APPROVED** [review]

**Review by Kevin-CB — APPROVED** [review]
I tested it locally and everything seems fine

And there are no more CSP warnings.

**Pull request merged**